src/main/java/com/google/sps/servlets/ProfileServlet.java

```diff
@@ -0,0 +1,121 @@
+        sessionDatastore = new RealTutorSessionDatastore();
+        availabilityDatastore = new RealAvailabilityDatastore();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+        
+        String userId = Optional.ofNullable(request.getParameter("user-id")).orElse(null);
+        
+        // Find out whether the user is a student or a tutor
+        Query query = new Query("User").setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
+        PreparedQuery results = datastore.prepare(query);
+        Entity entity = results.asSingleEntity();
+        String role = (String) entity.getProperty("role"); 
+
+        // User is a student, get their info
+        if (role.toLowerCase().equals("student")) {
+            query = new Query("Student").setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
+            results = datastore.prepare(query);
+            entity = results.asSingleEntity();
+
+            String name = (String) entity.getProperty("name");
+            String bio = (String) entity.getProperty("bio");
+            String pfp = (String) entity.getProperty("pfp");
+            String email = (String) entity.getProperty("email");
+            ArrayList<String> learning = (ArrayList) entity.getProperty("learning");
+            ArrayList<TutorSession> scheduledSessions = (ArrayList) sessionDatastore.getScheduledSessionsForStudent(email);
+
+            Student student = new Student(name, bio, pfp, email, learning, scheduledSessions);
+
+            String json = new Gson().toJson(student);
+            response.getWriter().println(json);
+        } else {   // User is a tutor, get their info
+            query = new Query("Tutor").setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
+            results = datastore.prepare(query);
+            entity = results.asSingleEntity();
+
+            String name = (String) entity.getProperty("name");
+            String bio = (String) entity.getProperty("bio");
+            String pfp = (String) entity.getProperty("pfp");
+            String email = (String) entity.getProperty("email");
+            ArrayList<String> skills = (ArrayList) entity.getProperty("topics");
+            ArrayList<TimeRange> availability = (ArrayList) availabilityDatastore.getAvailabilityForTutor(email);
+            ArrayList<TutorSession> scheduledSessions = (ArrayList) sessionDatastore.getScheduledSessionsForTutor(email);
+
+            Tutor tutor = new Tutor(name, bio, pfp, email, skills, availability, scheduledSessions);
+
+            String json = new Gson().toJson(tutor);
+            response.getWriter().println(json);
+        }
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    }
```

Comment:
Would this method be used to edit/update a user's bio and profile picture or do you envision that being elsewhere?

Reply:
I was planning to have the editing option here